pkg/render/prometheus_api.go

```diff
@@ -0,0 +1,269 @@
+}
+
+// calicoNodePrometheusService sets up a service for the Prometheus Service/Proxy deployment
+func (p *tigeraPrometheusAPIComponent) calicoNodePrometheusService() *corev1.Service {
+
+	s := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      calicoNodePrometheusServiceName,
+			Namespace: common.TigeraPrometheusNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"k8s-app": tigeraPrometheusAPIName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "web",
+					Port:       common.PrometheusDefaultPort,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(p.prometheusServiceListenPort),
+				},
+			},
+		},
+	}
+
+	return s
+}
+
+// tigeraPrometheusServiceDeployment deployment for the Prometheus Service/Proxy pod and image
+func (p *tigeraPrometheusAPIComponent) tigeraPrometheusServiceDeployment() *appsv1.Deployment {
+	var replicas int32 = 1
+	podDnsPolicy := corev1.DNSClusterFirst
+	podHostNetworked := false
+
+	// set to host networked if cluster is on EKS using Calico CNI since the EKS managed Kubernetes APIserver
+	// cannot reach the pod network in this configuration. This is because Calico cannot manage
+	// the managed control plane nodes' network
+	if p.installation.KubernetesProvider == operator.ProviderEKS &&
+		p.installation.CNI.Type == operator.PluginCalico {
+		podHostNetworked = true
+		// corresponding dns policy for hostnetwoked pods to resolve the service hostname urls
+		podDnsPolicy = corev1.DNSClusterFirstWithHostNet
+	}
+
+	d := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tigeraPrometheusAPIName,
+			Namespace: common.TigeraPrometheusNamespace,
+			Labels: map[string]string{
+				"k8s-app": tigeraPrometheusAPIName,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k8s-app": tigeraPrometheusAPIName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      tigeraPrometheusAPIName,
+					Namespace: common.TigeraPrometheusNamespace,
+					Labels: map[string]string{
+						"k8s-app": tigeraPrometheusAPIName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					DNSPolicy:        podDnsPolicy,
+					HostNetwork:      podHostNetworked,
+					ImagePullSecrets: secret.GetReferenceList(p.pullSecrets),
+					NodeSelector:     p.installation.ControlPlaneNodeSelector,
+					Tolerations:      []corev1.Toleration{rmeta.TolerateMaster},
+					Containers: []corev1.Container{
+						p.prometheusServiceContainers(),
+					},
+				},
+			},
+		},
+	}
+
+	return d
+}
+
+func (p *tigeraPrometheusAPIComponent) tigeraPrometheusApiPodSecurityPolicy() *policyv1beta1.PodSecurityPolicy {
```

Comment:
similar comment here about dropping `tigeraPrometheus` from the function name.

Reply:
https://github.com/tigera/operator/pull/1379/commits/0aecaf91feb429b1e0bf6e1e0d25be76e044121e

Reply:
Same here, I think you should shorten it to `podSecurityPolicy`